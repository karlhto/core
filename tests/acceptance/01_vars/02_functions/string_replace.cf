
#######################################################
#
# Test string_replace function
#
#######################################################

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent test
{
  vars:
      "test" string => "abcdefghij";

      # replace the empty string with empty string, globally
      "void" string => string_replace($(test), "", "");
      # replace the empty string with 'x', globally (matches on character boundaries)
      "void1" string => string_replace($(test), "", "x");

      # no matches, original intact
      "nomatch" string => string_replace($(test), "nonesuch", "TEST");

      # empty cases
      "empty1" string => string_replace("", "abc", "ABC");
      "empty2" string => string_replace("", "abc", "");
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
