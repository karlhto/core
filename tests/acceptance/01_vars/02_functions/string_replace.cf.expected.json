{
  "empty1": "",
  "empty2": "",
  "nomatch": "abcdefghij",
  "test": "abcdefghij",
  "void": "abcdefghij",
  "void1": "abcdefghij"
}
